stats.cpp

```diff
@@ -1,5 +1,32 @@
 #include "stats.h"
-
-Stats Statistics::ComputeStatistics(const std::vector<___>& ) {
-    //Implement statistics here
+#include <bits/stdc++.h>
+#include <math.h>
+namespace Statistics {
+    
+    std::vector<float> RemoveNANElements(const std::vector<float>& floatVector) {
+        std::vector<float> floatVectorWithoutNAN = {};
+        std::vector<float>::const_iterator itr = floatVector.begin();
+        while(itr != floatVector.end())
+        {
+            if(!isnan(*itr))
+                floatVectorWithoutNAN.push_back(*itr);
+            itr++;
+        }
+        return floatVectorWithoutNAN;
+    }
+    
+    Stats ComputeStatistics(const std::vector<float>& floatVector) {
+        struct Stats statsObj = {NAN, NAN, NAN};
+        if(floatVector.size() == 0)
```

Comment:
We could remove this check... Make the size-check only once, after the RemoveNANElements. Most of the time, the values will be valid. By repeating the size check, the overhead of a double check is incurred on them.

Reply:
Yes! Thanks a lot, I didn't realise that. The complexity of ComputeStatistics also reduced from 3 to 2!